core/mutation/mutation.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mutation implements the monitor service. This package contains the
+// core functionality.
+package mutation
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/google/keytransparency/core/mutator"
+	"github.com/google/keytransparency/core/transaction"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	tpb "github.com/google/keytransparency/core/proto/keytransparency_v1_types"
+	"github.com/google/trillian"
+)
+
+// Server holds internal state for the monitor server core functionality.
+type Server struct {
+	logID     int64
+	mapID     int64
+	tlog      trillian.TrillianLogClient
+	tmap      trillian.TrillianMapClient
+	mutations mutator.Mutation
+	factory   transaction.Factory
+}
+
+// New creates a new instance of the monitor server.
+func New(logID int64,
+	mapID int64,
+	tlog trillian.TrillianLogClient,
+	tmap trillian.TrillianMapClient,
+	mutations mutator.Mutation,
+	factory transaction.Factory) *Server {
+	return &Server{
+		logID:     logID,
+		mapID:     mapID,
+		tlog:      tlog,
+		tmap:      tmap,
+		mutations: mutations,
+		factory:   factory,
+	}
+}
+
+// GetMutations returns a list of mutations paged by epoch number.
+func (s *Server) GetMutations(ctx context.Context, in *tpb.GetMutationsRequest) (*tpb.GetMutationsResponse, error) {
+	// Get signed map root by revision.
+	resp, err := s.tmap.GetSignedMapRootByRevision(ctx, &trillian.GetSignedMapRootByRevisionRequest{
+		MapId:    s.mapID,
+		Revision: in.Epoch,
+	})
+	if err != nil {
+		log.Printf("GetSignedMapRootByRevision(%v, %v): %v", s.mapID, in.Epoch, err)
+		return nil, grpc.Errorf(codes.Internal, "Get signed map root failed")
+	}
+
+	// Get highest and lowest sequence number.
+	highestSeq := uint64(resp.GetMapRoot().GetMetadata().HighestFullyCompletedSeq)
```

Comment:
Done.

However, I don't think this will be a problem. Generally, we're using `Get` for all non-trivial types (e.g., integers) and since Trillian uses `proto3`, any missing field will have a default value (there are no `Has` methods in `proto3` AFAIK). If you think this might still cause a problem on the long run, I'll start using `Get` everywhere and gradually change the codebase in that direction.

Using `Get` everywhere might make the code look nice and be consistent.